### PR TITLE
NP-9098: handler for initializing default roles

### PR DESCRIPTION
**Pull request by axthosarouris**
This is a bit of two PRs in one. One PR is the creation of a handler for initialiazing the default roles. The other one is using a Set of AccessRights everywhere instead of List of Strings or List of AccessRights

### CODE REVIEW

File: user-access-handlers/src/main/java/no/unit/nva/handlers/IdentityServiceInitHandler.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package no.unit.nva.handlers;
```
Comment (axthosarouris) [inline user-access-handlers/src/main/java/no/unit/nva/handlers/IdentityServiceInitHandler.java:1]: This is the initialization handler.

File: user-access-handlers/src/main/java/no/unit/nva/handlers/models/RoleList.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package no.unit.nva.handlers.models;
```
Comment (axthosarouris) [inline user-access-handlers/src/main/java/no/unit/nva/handlers/models/RoleList.java:1]: This is the response of the handler.

File: user-access-handlers/src/test/java/no/unit/nva/handlers/EntityUtils.java [hunk 3]
```diff
@@ -18,14 +20,11 @@ public final class EntityUtils {
 
     public static final URI SOME_INSTITUTION = randomCristinOrgId();
     public static final String EMPTY_STRING = "";
-    public static final Set<String> SAMPLE_ACCESS_RIGHTS =
-        Collections.singleton("APPROVE_DOI_REQUEST");
+    public static final Set<AccessRight> SAMPLE_ACCESS_RIGHTS =
```
Comment (axthosarouris) [inline user-access-handlers/src/test/java/no/unit/nva/handlers/EntityUtils.java:23]: Use AccessRight for AccessRights

File: user-access-handlers/src/test/java/no/unit/nva/handlers/IdentityServiceInitHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import no.unit.nva.handlers.models.RoleList;
+import no.unit.nva.stubs.FakeContext;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import no.unit.nva.useraccessservice.model.RoleDto;
+import nva.commons.apigateway.AccessRight;
+import nva.commons.apigateway.GatewayResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class IdentityServiceInitHandlerTest extends LocalIdentityService {
```
Comment (axthosarouris) [inline user-access-handlers/src/test/java/no/unit/nva/handlers/IdentityServiceInitHandlerTest.java:25]: This is the test for the init handler

File: user-access-internal-model/src/main/java/no/unit/nva/useraccessservice/dao/AccessRightSetConverterProvider.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package no.unit.nva.useraccessservice.dao;
+
+import java.util.Set;
+import nva.commons.apigateway.AccessRight;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeConverterProvider;
+import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
+
+public class AccessRightSetConverterProvider implements AttributeConverterProvider {
```
Comment (axthosarouris) [inline user-access-internal-model/src/main/java/no/unit/nva/useraccessservice/dao/AccessRightSetConverterProvider.java:9]: Necessary class for using the AccessRightSetConverter

File: user-access-internal-model/src/main/java/no/unit/nva/useraccessservice/dao/RoleDb.java [hunk 6]
```diff
@@ -19,15 +16,15 @@
...
 import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
 import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute;
 import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
-import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbIgnore;
 import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbIgnoreNulls;
 import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
 import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey;
 
-@DynamoDbBean
+@DynamoDbBean(converterProviders = {AccessRightSetConverterProvider.class, DefaultAttributeConverterProvider.class})
```
Comment (axthosarouris) [inline user-access-internal-model/src/main/java/no/unit/nva/useraccessservice/dao/RoleDb.java:27]: Here we employ the AccessRightConverter

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: user-access-handlers/src/main/java/no/unit/nva/handlers/IdentityServiceInitHandler.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            .withAccessRights(
+                List.of(APPROVE_DOI_REQUEST, REJECT_DOI_REQUEST, READ_DOI_REQUEST, EDIT_OWN_INSTITUTION_RESOURCES))
+            .build();
+        var institutionAdmin = RoleDto.newBuilder().withRoleName("Institution-admin")
+            .withAccessRights(List.of(EDIT_OWN_INSTITUTION_RESOURCES,
+                                      EDIT_OWN_INSTITUTION_PROJECTS,
+                                      EDIT_OWN_INSTITUTION_USERS))
+            .build();
+        var applicationAdmin = RoleDto.newBuilder().withRoleName("App-admin")
+            .withAccessRights(List.of(ADMINISTRATE_APPLICATION))
```
[the remark this answers is not in the document]
Comment (axthosarouris) [reply user-access-handlers/src/main/java/no/unit/nva/handlers/IdentityServiceInitHandler.java:55]: this is in commons

**Pull request merged**